diff --git a/dv/env/kmac_env.sv b/dv/env/kmac_env.sv
index 93c5c02c..c78784b8 100644
--- a/dv/env/kmac_env.sv
+++ b/dv/env/kmac_env.sv
@@ -25,9 +25,9 @@ class kmac_env extends cip_base_env #(
     end
 
     // get ext interfaces
-    keymgr_sideload_agent = key_sideload_agent#(keymgr_pkg::hw_key_req_t)::type_id::create(
+    keymgr_sideload_agent = key_sideload_agent#(kmac_pkg::hw_key_req_t)::type_id::create(
       "keymgr_sideload_agent", this);
-    uvm_config_db#(key_sideload_agent_cfg#(keymgr_pkg::hw_key_req_t))::set(
+    uvm_config_db#(key_sideload_agent_cfg#(kmac_pkg::hw_key_req_t))::set(
       this, "keymgr_sideload_agent*", "cfg", cfg.keymgr_sideload_agent_cfg);
 
     // config kmac virtual interface
diff --git a/dv/env/kmac_env_cfg.sv b/dv/env/kmac_env_cfg.sv
index efbb3905..bbd7bbba 100644
--- a/dv/env/kmac_env_cfg.sv
+++ b/dv/env/kmac_env_cfg.sv
@@ -61,7 +61,7 @@ class kmac_env_cfg extends cip_base_env_cfg #(.RAL_T(kmac_reg_block));
       m_kmac_app_agent_cfg[i] = kmac_app_agent_cfg::type_id::create(name);
       m_kmac_app_agent_cfg[i].if_mode = dv_utils_pkg::Host;
     end
-    keymgr_sideload_agent_cfg = key_sideload_agent_cfg#(keymgr_pkg::hw_key_req_t)::type_id
+    keymgr_sideload_agent_cfg = key_sideload_agent_cfg#(kmac_pkg::hw_key_req_t)::type_id
                                 ::create("keymgr_sideload_agent_cfg");
     void'($value$plusargs("enable_masking=%0d", enable_masking));
     void'($value$plusargs("sw_key_masked=%0d", sw_key_masked));
diff --git a/dv/env/kmac_env_cov.sv b/dv/env/kmac_env_cov.sv
index 18206d81..6b1897fc 100644
--- a/dv/env/kmac_env_cov.sv
+++ b/dv/env/kmac_env_cov.sv
@@ -125,7 +125,7 @@ endgroup
 class app_cg_wrap;
   // done signal is sent while Keccak rounds are running
   covergroup app_cg(string name) with function sample(bit single_beat,
-                                                      bit [keymgr_pkg::KmacDataIfWidth/8-1:0] strb,
+                                                      bit [kmac_pkg::MsgWidth/8-1:0] strb,
                                                       bit err,
                                                       bit is_done,
                                                       bit in_keccak);
@@ -177,11 +177,11 @@ class app_cg_wrap;
 
   function new(string name = "app_cg");
     app_cg = new(name);
-    app_cfg_reg_cg = new(name);
+    app_cfg_reg_cg = new({name, "_cfg_reg"});
   endfunction
 
   function void app_sample(bit single_beat,
-                           bit [keymgr_pkg::KmacDataIfWidth/8-1:0] strb,
+                           bit [kmac_pkg::MsgWidth/8-1:0] strb,
                            bit err,
                            bit is_done,
                            bit in_keccak);
diff --git a/dv/env/kmac_env_pkg.sv b/dv/env/kmac_env_pkg.sv
index 3cf6e591..88ebe476 100644
--- a/dv/env/kmac_env_pkg.sv
+++ b/dv/env/kmac_env_pkg.sv
@@ -19,7 +19,6 @@ package kmac_env_pkg;
   import kmac_app_agent_pkg::*;
   import kmac_ral_pkg::*;
   import kmac_pkg::*;
-  import keymgr_pkg::*;
   import key_sideload_agent_pkg::*;
   import sec_cm_pkg::*;
 
diff --git a/dv/env/kmac_scoreboard.sv b/dv/env/kmac_scoreboard.sv
index a6222f68..018009c7 100644
--- a/dv/env/kmac_scoreboard.sv
+++ b/dv/env/kmac_scoreboard.sv
@@ -110,8 +110,8 @@ class kmac_scoreboard extends cip_base_scoreboard #(
   // key length enum
   key_len_e key_len;
 
-  bit [keymgr_pkg::KmacDataIfWidth-1:0]   kmac_app_block_data;
-  bit [keymgr_pkg::KmacDataIfWidth/8-1:0] kmac_app_block_strb;
+  bit [kmac_pkg::MsgWidth-1:0]   kmac_app_block_data;
+  bit [kmac_pkg::MsgWidth/8-1:0] kmac_app_block_strb;
   int kmac_app_block_strb_size = 0;
   bit kmac_app_last;
 
@@ -263,13 +263,13 @@ class kmac_scoreboard extends cip_base_scoreboard #(
 
             // Once valid sideload keys have been seen, update scoreboard state.
             //
-            // Note: max size of sideloaded key is keymgr_pkg::KeyWidth
+            // Note: max size of sideloaded key is kmac_pkg::KeyWidth
 
             sideload_key = cfg.keymgr_sideload_agent_cfg.vif.sideload_key;
 
             `uvm_info(`gfn, $sformatf("detected valid sideload_key: %0p", sideload_key), UVM_HIGH)
 
-            for (int i = 0; i < keymgr_pkg::KeyWidth / 32; i++) begin
+            for (int i = 0; i < kmac_pkg::KeyWidth / 32; i++) begin
               keymgr_keys[0][i] = sideload_key.key[0][i*32 +: 32];
               keymgr_keys[1][i] = sideload_key.key[1][i*32 +: 32];
             end
@@ -287,12 +287,12 @@ class kmac_scoreboard extends cip_base_scoreboard #(
   // Get sideload keys, pack and return the keys.
   virtual function bit [KMAC_NUM_SHARES-1:0][KMAC_NUM_KEYS_PER_SHARE-1:0][31:0] get_keymgr_keys();
     bit [KMAC_NUM_SHARES-1:0][KMAC_NUM_KEYS_PER_SHARE-1:0][31:0] keymgr_keys;
-    keymgr_pkg::hw_key_req_t sideload_key;
+    kmac_pkg::hw_key_req_t sideload_key;
 
     if (cfg.keymgr_sideload_agent_cfg.vif.sideload_key.valid) begin
       sideload_key = cfg.keymgr_sideload_agent_cfg.vif.sideload_key;
       `uvm_info(`gfn, $sformatf("get valid sideload_key: %0p", sideload_key), UVM_HIGH)
-      for (int i = 0; i < keymgr_pkg::KeyWidth / 32; i++) begin
+      for (int i = 0; i < kmac_pkg::KeyWidth / 32; i++) begin
         keymgr_keys[0][i] = sideload_key.key[0][i*32 +: 32];
         keymgr_keys[1][i] = sideload_key.key[1][i*32 +: 32];
       end
@@ -561,7 +561,7 @@ class kmac_scoreboard extends cip_base_scoreboard #(
                 // sample coverage
                 if (cfg.en_cov) begin
                   cov.app_cg_wrappers[app_mode].app_sample(
-                    kmac_app_rsp.byte_data_q.size() <= keymgr_pkg::KmacDataIfWidth/8,
+                    kmac_app_rsp.byte_data_q.size() <= kmac_pkg::MsgWidth/8,
                     '0,
                     kmac_app_rsp.rsp_error,
                     1,
diff --git a/dv/env/seq_lib/kmac_smoke_vseq.sv b/dv/env/seq_lib/kmac_smoke_vseq.sv
index ddb1021b..70ee26ba 100644
--- a/dv/env/seq_lib/kmac_smoke_vseq.sv
+++ b/dv/env/seq_lib/kmac_smoke_vseq.sv
@@ -103,8 +103,8 @@ class kmac_smoke_vseq extends kmac_base_vseq;
   // Do a full message hash, repeated num_trans times
   task body();
 
-    logic [keymgr_pkg::KeyWidth-1:0] sideload_share0;
-    logic [keymgr_pkg::KeyWidth-1:0] sideload_share1;
+    logic [kmac_pkg::KeyWidth-1:0] sideload_share0;
+    logic [kmac_pkg::KeyWidth-1:0] sideload_share1;
     key_sideload_set_seq sideload_seq;
 
     `uvm_info(`gfn, $sformatf("Starting %0d message hashes", num_trans), UVM_LOW)
diff --git a/dv/tb.sv b/dv/tb.sv
index 3c145ca1..ac702a7c 100644
--- a/dv/tb.sv
+++ b/dv/tb.sv
@@ -17,7 +17,7 @@ module tb;
   wire clk, rst_n, rst_shadowed_n;
   wire [NUM_MAX_INTERRUPTS-1:0] interrupts;
   // keymgr/kmac sideload wires
-  keymgr_pkg::hw_key_req_t kmac_sideload_key;
+  kmac_pkg::hw_key_req_t kmac_sideload_key;
   // kmac_app interfaces
   kmac_pkg::app_req_t [NUM_APP_INTF-1:0] app_req;
   kmac_pkg::app_rsp_t [NUM_APP_INTF-1:0] app_rsp;
