diff --git a/key_sideload_agent/key_sideload_agent.core b/key_sideload_agent/key_sideload_agent.core
index 72a73045..45332fa9 100644
--- a/key_sideload_agent/key_sideload_agent.core
+++ b/key_sideload_agent/key_sideload_agent.core
@@ -9,7 +9,7 @@ filesets:
     depend:
       - lowrisc:dv:dv_utils
       - lowrisc:dv:dv_lib
-      - lowrisc:ip:keymgr_pkg
+      - lowrisc:ip:kmac_pkg
     files:
       - key_sideload_if.sv
       - key_sideload_agent_pkg.sv
diff --git a/key_sideload_agent/key_sideload_agent.sv b/key_sideload_agent/key_sideload_agent.sv
index 1f6e3c80..b9e35257 100644
--- a/key_sideload_agent/key_sideload_agent.sv
+++ b/key_sideload_agent/key_sideload_agent.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_agent#(
-  parameter type KEY_T = keymgr_pkg::hw_key_req_t
+  parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends dv_base_agent #(
   .CFG_T          (key_sideload_agent_cfg#(KEY_T)),
   .DRIVER_T       (key_sideload_driver#(KEY_T)),
diff --git a/key_sideload_agent/key_sideload_agent_cfg.sv b/key_sideload_agent/key_sideload_agent_cfg.sv
index 9726e0c9..3423a21b 100644
--- a/key_sideload_agent/key_sideload_agent_cfg.sv
+++ b/key_sideload_agent/key_sideload_agent_cfg.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_agent_cfg #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends dv_base_agent_cfg;
 
 
diff --git a/key_sideload_agent/key_sideload_agent_cov.sv b/key_sideload_agent/key_sideload_agent_cov.sv
index 32ec149e..c3a50e77 100644
--- a/key_sideload_agent/key_sideload_agent_cov.sv
+++ b/key_sideload_agent/key_sideload_agent_cov.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_agent_cov #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends dv_base_agent_cov #(key_sideload_agent_cfg#(KEY_T));
   `uvm_component_param_utils(key_sideload_agent_cov#(KEY_T))
 
diff --git a/key_sideload_agent/key_sideload_driver.sv b/key_sideload_agent/key_sideload_driver.sv
index 2ffb699a..4df961ac 100644
--- a/key_sideload_agent/key_sideload_driver.sv
+++ b/key_sideload_agent/key_sideload_driver.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_driver#(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends dv_base_driver #(.ITEM_T(key_sideload_item#(KEY_T)),
                            .CFG_T (key_sideload_agent_cfg#(KEY_T)));
   `uvm_component_param_utils(key_sideload_driver#(KEY_T))
diff --git a/key_sideload_agent/key_sideload_if.sv b/key_sideload_agent/key_sideload_if.sv
index 1f8b521e..31281c1b 100644
--- a/key_sideload_agent/key_sideload_if.sv
+++ b/key_sideload_agent/key_sideload_if.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 interface key_sideload_if #(
-  parameter type KEY_T = keymgr_pkg::hw_key_req_t
+  parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) (
   input logic clk_i,
   input logic rst_ni,
diff --git a/key_sideload_agent/key_sideload_item.sv b/key_sideload_agent/key_sideload_item.sv
index 7754a65c..961d6863 100644
--- a/key_sideload_agent/key_sideload_item.sv
+++ b/key_sideload_agent/key_sideload_item.sv
@@ -3,9 +3,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_item #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends uvm_sequence_item;
-  localparam int KeyWidth = ($bits(KEY_T)-1)/keymgr_pkg::Shares;
+  localparam int KeyWidth = ($bits(KEY_T)-1)/kmac_pkg::Shares;
 
   // random variables
   rand bit                valid;
diff --git a/key_sideload_agent/key_sideload_monitor.sv b/key_sideload_agent/key_sideload_monitor.sv
index d5856895..a025b4fe 100644
--- a/key_sideload_agent/key_sideload_monitor.sv
+++ b/key_sideload_agent/key_sideload_monitor.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_monitor #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends dv_base_monitor #(
     .ITEM_T (key_sideload_item#(KEY_T)),
     .CFG_T  (key_sideload_agent_cfg#(KEY_T)),
diff --git a/key_sideload_agent/key_sideload_sequencer.sv b/key_sideload_agent/key_sideload_sequencer.sv
index a9412175..4e37dc4d 100644
--- a/key_sideload_agent/key_sideload_sequencer.sv
+++ b/key_sideload_agent/key_sideload_sequencer.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_sequencer #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 )  extends dv_base_sequencer #(
     .ITEM_T (key_sideload_item#(KEY_T)),
     .CFG_T  (key_sideload_agent_cfg#(KEY_T))
diff --git a/key_sideload_agent/seq_lib/key_sideload_base_seq.sv b/key_sideload_agent/seq_lib/key_sideload_base_seq.sv
index 60def448..147c0f74 100644
--- a/key_sideload_agent/seq_lib/key_sideload_base_seq.sv
+++ b/key_sideload_agent/seq_lib/key_sideload_base_seq.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_base_seq #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends dv_base_seq #(
     .REQ         (key_sideload_item#(KEY_T)),
     .CFG_T       (key_sideload_agent_cfg#(KEY_T)),
diff --git a/key_sideload_agent/seq_lib/key_sideload_set_seq.sv b/key_sideload_agent/seq_lib/key_sideload_set_seq.sv
index b25bfb25..79857b60 100644
--- a/key_sideload_agent/seq_lib/key_sideload_set_seq.sv
+++ b/key_sideload_agent/seq_lib/key_sideload_set_seq.sv
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 class key_sideload_set_seq #(
-    parameter type KEY_T = keymgr_pkg::hw_key_req_t
+    parameter type KEY_T = kmac_pkg::hw_key_req_t
 ) extends key_sideload_base_seq #(KEY_T);
 
   `uvm_object_utils(key_sideload_set_seq#(KEY_T))
diff --git a/kmac_app_agent/kmac_app_agent.core b/kmac_app_agent/kmac_app_agent.core
index 79527ccb..acaf70c5 100644
--- a/kmac_app_agent/kmac_app_agent.core
+++ b/kmac_app_agent/kmac_app_agent.core
@@ -10,7 +10,6 @@ filesets:
       - lowrisc:dv:dv_utils
       - lowrisc:dv:dv_lib
       - lowrisc:dv:push_pull_agent
-      - lowrisc:ip:keymgr_pkg
       - lowrisc:ip:kmac_pkg
     files:
       - kmac_app_agent_pkg.sv
diff --git a/kmac_app_agent/kmac_app_agent_pkg.sv b/kmac_app_agent/kmac_app_agent_pkg.sv
index 6af1f15d..4da79b45 100644
--- a/kmac_app_agent/kmac_app_agent_pkg.sv
+++ b/kmac_app_agent/kmac_app_agent_pkg.sv
@@ -8,7 +8,7 @@ package kmac_app_agent_pkg;
   import dv_utils_pkg::*;
   import dv_base_agent_pkg::*;
   import dv_lib_pkg::*;
-  import keymgr_pkg::*;
+  import kmac_pkg::*;
   import push_pull_agent_pkg::*;
 
   // macro includes
@@ -16,9 +16,10 @@ package kmac_app_agent_pkg;
   `include "dv_macros.svh"
 
   // parameters
-  parameter int KMAC_REQ_DATA_WIDTH = keymgr_pkg::KmacDataIfWidth       // data width
-                                      + keymgr_pkg::KmacDataIfWidth / 8 // data mask width
-                                      + 1;                              // bit last
+  parameter int KmacDataIfWidth = kmac_pkg::MsgWidth;
+  parameter int KMAC_REQ_DATA_WIDTH = KmacDataIfWidth       // data width
+                                      + KmacDataIfWidth / 8 // data mask width
+                                      + 1;                  // bit last
 
   `define CONNECT_DATA_WIDTH .HostDataWidth(kmac_app_agent_pkg::KMAC_REQ_DATA_WIDTH)
 
diff --git a/kmac_app_agent/kmac_app_intf.sv b/kmac_app_agent/kmac_app_intf.sv
index 8fe2c411..b5c926cb 100644
--- a/kmac_app_agent/kmac_app_intf.sv
+++ b/kmac_app_agent/kmac_app_intf.sv
@@ -5,8 +5,6 @@
 // verilog_lint: waive interface-name-style
 interface kmac_app_intf (input clk, input rst_n);
 
-  import keymgr_pkg::*;
-
   dv_utils_pkg::if_mode_e if_mode; // interface mode - Host or Device
 
   // interface pins used to connect with DUT
@@ -60,7 +58,7 @@ interface kmac_app_intf (input clk, input rst_n);
           clk, !rst_n || if_mode == dv_utils_pkg::Host)
 
   // Check strb is aligned to LSB, for example: if strb[1]==0, strb[$:2] should be 0 too
-  for (genvar k = 1; k < KmacDataIfWidth / 8 - 1; k++) begin : gen_strb_check
+  for (genvar k = 1; k < kmac_app_agent_pkg::KmacDataIfWidth / 8 - 1; k++) begin : gen_strb_check
     `ASSERT(StrbAlignLSB_A, kmac_data_req.valid && kmac_data_req.strb[k] === 0 |->
                             kmac_data_req.strb[k+1] === 0,
                             clk, !rst_n || if_mode == dv_utils_pkg::Host)
