diff --git a/dv/env/pwrmgr_if.sv b/dv/env/pwrmgr_if.sv
index 00ec532..5f6ace2 100644
--- a/dv/env/pwrmgr_if.sv
+++ b/dv/env/pwrmgr_if.sv
@@ -87,12 +87,8 @@ interface pwrmgr_if (
   always_comb lowpwr_cfg_wen = `PATH_TO_DUT.lowpwr_cfg_wen;
 
   // reset status
-  always_comb
-    reset_status = {
-      `PATH_TO_DUT.u_reg.reset_status_val_1_qs, `PATH_TO_DUT.u_reg.reset_status_val_0_qs
-    };
-  always_comb
-    reset_en_q = {`PATH_TO_DUT.u_reg.reset_en_en_1_qs, `PATH_TO_DUT.u_reg.reset_en_en_0_qs};
+  always_comb reset_status = `PATH_TO_DUT.u_reg.reset_status_qs;
+  always_comb reset_en_q = `PATH_TO_DUT.u_reg.reset_en_qs;
   always_comb
     wakeup_en = {
       `PATH_TO_DUT.reg2hw.wakeup_en[5].q,
